src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -2614,11 +2615,41 @@ internal static int IndexOfAnyValueType<T>(ref T searchSpace, T value0, T value1
             return -1;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveOptimization)]
+        private static bool VectorContainsMatch<TValue>(Vector128<TValue> inputVector) where TValue : struct, INumber<TValue>
+        {
+            // The inputVector has all the bits of a matching element set.
+            if (AdvSimd.Arm64.IsSupported)
+            {
+                // We select a larger byte from the adjacent pair of bytes. Thus, if any of the 16 bytes of inputVector are set,
+                // representing a matched element, they get reflected in the bottom half (8 bytes) of the register.
+                // We then detect a match when any of the bottom 8 bytes are non-zero.
+                return AdvSimd.Arm64.MaxPairwise(inputVector.AsByte(), inputVector.AsByte()).AsUInt64().ToScalar() != 0;
+            }
+            else
+            {
+                return (inputVector != Vector128<TValue>.Zero);
+            }
+        }
+
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static int ComputeFirstIndex<T>(ref T searchSpace, ref T current, Vector128<T> equals) where T : struct
         {
-            uint notEqualsElements = equals.ExtractMostSignificantBits();
-            int index = BitOperations.TrailingZeroCount(notEqualsElements);
+            int index;
+            if (AdvSimd.Arm64.IsSupported && Unsafe.SizeOf<T>() > 1)
+            {
+                // PairwiseMax selects a larger of the adjacent elements. Now the bottom half of
+                // the target register (64 bits) represents elements of half the original size.
+                // Dividing the number of trailing zeros by 4 gives the byte offset position of
+                // the first matching element, and dividing it by the element size gives its index.
+                index = (BitOperations.TrailingZeroCount(
```

Comment:
Should be a nice improvements, although we wanted to handle `ExtractMostSignificantBits` in JIT for arm64 to this when we see that input is a result of a comparison cc @tannergooding

Reply:
Alright. That makes sense. I will close this PR in this case.